packages/sdk/src/oracle/services/erc20.ts

```diff
@@ -0,0 +1,43 @@
+import { erc20 } from "../../clients";
+import Multicall2 from "../../multicall2";
+import { BatchReadWithErrors, BatchReadWithErrorsType, Calls } from "../../utils";
+import { BigNumber, Provider } from "../types/ethers";
+
+const batchProps: Calls = [["symbol"], ["name"], ["decimals"], ["totalSupply"]];
+type Props = {
+  symbol: string;
+  name: string;
+  decimals: number;
+  totalSupply: BigNumber;
+};
+export class Erc20 {
```

Comment:
Why did you create two classes Erc20 and Erc20 multicall. Why not just make one class that includes all the functions? If for modularity reasons, I'm having trouble seeing the separation between the two? They both offer a client to the ERC20 token, one just seems to add a little dynamism, right?

Reply:
the multicall erc20 extends from the plain erc20, overriding the getProps function. Given that the app is multichain and we dont know if a chain has multicall deployed, i figured we would want a non multicall option. The mutlicall option simply needs to override calls on the base class to optimize.

Reply:
Oh interesting. Do we assume multicall exists everywhere in other parts of the SDK?

I just thought we assumed this existed everywhere. Seems reasonable to leave a non-multicall option if that's what we do elsewhere.

Reply:
We learned from the API that when deploying to polygon, we couldn't assume multicall existed. Our solution in that case was to deploy our own multicall contract. So if i know now that the app will run on multi chain, i dont make that assumption anymore.  The only other place multicall is used is the across pool client, which its assumed the liquidity pools are on mainnet.